nixos/modules/services/x11/clight.nix

```diff
@@ -0,0 +1,149 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.clight;
+  clightConf = pkgs.writeText "clight.conf" cfg.extraConfig;
+in {
+  options.services.clight = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable clight or not.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.clight;
+      defaultText = "pkgs.clight";
+      description = ''
+        Clight derivation to use.
+      '';
+    };
+
+    clightd = mkOption {
+      type = types.package;
+      default = pkgs.clightd;
```

Comment:
I'm not sure if it makes sense to have options for this, considering that there's only one version of clight, which can be overridden with overlays anyways. But people have differing opinions on this (see https://github.com/NixOS/nixpkgs/issues/50476). We can always add such options later if the need arises though. 2 options for this, which probably nobody ever uses, is a bit overkill imo